File: First Unique Character in a String.java
class Solution {
    public int firstUniqChar(String s) {
        if (s.length() == 0 | s == null){
            return -1;
        }
        
        Map<Character, Integer> freq = new HashMap<Character, Integer>();
        
        for (int i = 0; i < s.length(); i ++){
            char c = s.charAt(i);
            if (freq.containsKey(c)){
                
                freq.put(c,freq.get(c) + 1);
                
            } else {
                freq.put(c, 1);
            }
        }
        
        for (int i = 0; i < s.length(); i ++){
            char c = s.charAt(i);
            if (freq.get(c) == 1){
                return i;
            } 
        } 
        return -1;
    }
}

File: First Unique Character in a String.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        if len(s) == 0:
            return -1; 
        
        
        charFreq = {};
        for char in s:
            charFreq[char] = charFreq.get(char, 0) + 1;
            
            
        for i in range(len(s)):
            if charFreq[s[i]] != 1:
                continue;
            else:
                return i;
            
            
        return -1; 
        
        